src/NHibernate.Test/SystemTransactions/TransactionFixture.cs

```diff
@@ -29,11 +25,84 @@ public void CanUseSystemTransactionsToCommit()
 			using (ISession session = Sfi.OpenSession())
 			using (TransactionScope tx = new TransactionScope())
 			{
-				W w = session.Get<W>(identifier);
+				var w = session.Get<Person>(identifier);
 				Assert.IsNotNull(w);
 				session.Delete(w);
 				tx.Complete();
 			}
 		}
+
+		[Test]
+		public void FlushFromTransactionAppliesToDisposedSharingSession()
+		{
+			using (var s = OpenSession())
+			{
+				var builder = s.SessionWithOptions().Connection();
+
+				using (var t = new TransactionScope())
```

Comment:
At least it is not valid to have an entity graph with entities from different sessions. The valid approach testing that flush has happened would be to supply a listener or an interceptor.